netsim/devices/eos.py

```diff
@@ -101,6 +101,13 @@ def passive_stub_interfaces(node: Box, topology: Box) -> None:
 
     intf.ospf.network_type = 'point-to-point'
 
+def esi_identifier_format(node: Box, topology: Box) -> None:
+  # Arista EOS, in EVPN ESI, wants ESI in format: 0000:0000:0000:0000:0000
+  for intf in node.interfaces:
+    if intf.get('evpn._esi.id'):
+      esi = intf.evpn._esi.id.replace(":", "")
+      intf.evpn._esi._eos_id = ':'.join(esi[i:i+4] for i in range(0, len(esi), 4))
```

Comment:
This is a neat trick. I have to remember it ;)

Reply:
This looks more and more like "turning Python into Perl" contest 🤪 I was staring at your code for a few minutes, and still only guess that it works.